chia/_tests/core/mempool/test_mempool_manager.py

```diff
@@ -1985,21 +1986,26 @@ async def send_to_mempool(full_node: FullNodeSimulator, spend_bundle: SpendBundl
     async def fill_mempool_with_test_sbs(
         full_node_api: FullNodeSimulator,
     ) -> List[Tuple[bytes32, SerializedProgram, bytes32]]:
-        puzzle = SerializedProgram.to((1, [[ConditionOpcode.REMARK, b"1" * 12_000]]))
-        ph = puzzle.get_tree_hash()
-        for _ in range(3):
-            await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph))
-        coin_records = await full_node_api.full_node.coin_store.get_coin_records_by_puzzle_hash(False, ph)
-        coins = [cr.coin for cr in coin_records if cr.coin.amount == 250_000_000_000]
+        coins_and_puzzles = []
+        # Create different puzzles and use different (parent) coins to reduce
+        # the effects of block compression as much as possible.
+        for i in (1, 2):
+            puzzle = SerializedProgram.to((1, [[ConditionOpcode.REMARK, bytes([i] * 12_000)]]))
+            ph = puzzle.get_tree_hash()
+            for _ in range(2):
+                await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph))
+            coin_records = await full_node_api.full_node.coin_store.get_coin_records_by_puzzle_hash(False, ph)
+            coin = next(cr.coin for cr in coin_records if cr.coin.amount == 250_000_000_000)
+            coins_and_puzzles.append((coin, puzzle))
         sbs_info = []
-        for coin in coins:
+        for coin, puzzle in coins_and_puzzles:
             coin_spend = make_spend(coin, puzzle, SerializedProgram.to([]))
             sb = SpendBundle([coin_spend], G2Element())
             await send_to_mempool(full_node_api, sb)
             sbs_info.append((coin.name(), puzzle, sb.name()))
         return sbs_info
 
-    constants = test_constants.replace(MAX_BLOCK_COST_CLVM=max_block_clvm_cost)
+    constants = blockchain_constants.replace(MAX_BLOCK_COST_CLVM=max_block_clvm_cost)
```

Comment:
I take it this is the key change. previously we always tested the validation as if we were producing blocks for the pre-hard fork chain. is that right?

Reply:
Indeed.